src/lib/FlightTasks/tasks/FlightTaskAuto.cpp

```diff
@@ -97,99 +103,110 @@ bool FlightTaskAuto::_evaluateTriplets()
+			// Horizontal target is not finite.
+			_triplet_target(0) = _position(0);
+			_triplet_target(1) = _position(1);
+		}
 
-	// update all waypoints
-	_target = target;
+		if (!PX4_ISFINITE(_triplet_target(2))) {
+			_triplet_target(2) = _position(2);
+		}
 
-	if (!PX4_ISFINITE(_target(0)) || !PX4_ISFINITE(_target(1))) {
-		// Horizontal target is not finite. */
-		_target(0) = _position(0);
-		_target(1) = _position(1);
-	}
+		// If _triplet_target has updated, update also _triplet_prev_wp and _triplet_next_wp.
+		_prev_prev_wp = _triplet_prev_wp;
 
-	if (!PX4_ISFINITE(_target(2))) {
-		_target(2) = _position(2);
-	}
+		if (_isFinite(_sub_triplet_setpoint->get().previous) && _sub_triplet_setpoint->get().previous.valid) {
+			map_projection_project(&_reference_position, _sub_triplet_setpoint->get().previous.lat,
+					       _sub_triplet_setpoint->get().previous.lon, &_triplet_prev_wp(0), &_triplet_prev_wp(1));
+			_triplet_prev_wp(2) = -(_sub_triplet_setpoint->get().previous.alt - _reference_altitude);
+
+		} else {
+			_triplet_prev_wp = _position;
+		}
 
-	_prev_prev_wp = _prev_wp;
+		if (_type == WaypointType::loiter) {
+			_triplet_next_wp = _triplet_target;
 
-	if (_isFinite(_sub_triplet_setpoint->get().previous) && _sub_triplet_setpoint->get().previous.valid) {
-		map_projection_project(&_reference_position, _sub_triplet_setpoint->get().previous.lat,
-				       _sub_triplet_setpoint->get().previous.lon, &_prev_wp(0), &_prev_wp(1));
-		_prev_wp(2) = -(_sub_triplet_setpoint->get().previous.alt - _reference_altitude);
+		} else if (_isFinite(_sub_triplet_setpoint->get().next) && _sub_triplet_setpoint->get().next.valid) {
+			map_projection_project(&_reference_position, _sub_triplet_setpoint->get().next.lat,
+					       _sub_triplet_setpoint->get().next.lon, &_triplet_next_wp(0), &_triplet_next_wp(1));
+			_triplet_next_wp(2) = -(_sub_triplet_setpoint->get().next.alt - _reference_altitude);
 
-	} else {
-		_prev_wp = _position;
+		} else {
+			_triplet_next_wp = _triplet_target;
+		}
 	}
 
-	if (_type == WaypointType::loiter) {
-		_next_wp = _target;
+	// Check if yaw target is valid.
+	_yaw_setpoint = _sub_triplet_setpoint->get().current.yaw;
+
+	if (_type == WaypointType::follow_target && _sub_triplet_setpoint->get().current.yawspeed_valid) {
+		_yawspeed_setpoint = _sub_triplet_setpoint->get().current.yawspeed;
+		_yaw_setpoint = NAN;
+	}
 
-	} else if (_isFinite(_sub_triplet_setpoint->get().next) && _sub_triplet_setpoint->get().next.valid) {
-		map_projection_project(&_reference_position, _sub_triplet_setpoint->get().next.lat,
-				       _sub_triplet_setpoint->get().next.lon, &_next_wp(0), &_next_wp(1));
-		_next_wp(2) = -(_sub_triplet_setpoint->get().next.alt - _reference_altitude);
+	// Calculate the current vehicle state and check if it has updated.
+	State _previous_state = _current_state;
+	_current_state = _getCurrentState();
+	bool state_update = (_current_state != _previous_state) ? true : false;
```

Comment:
`bool state_update = _current_state != _previous_state` is enough.

I think you can use this check directly below in the if-statement (https://github.com/PX4/Firmware/pull/10006/files#diff-f4b91b66a95dd770c7f58b33f09f5943R208) instead of creating a boolean